x-pack/plugins/proxy/server/proxy_config.ts

```diff
@@ -0,0 +1,36 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { schema, TypeOf } from '@kbn/config-schema';
+import { Env } from 'src/core/server/config';
+
+const createProxySchema = schema.object({
+  updateInterval: schema.number(),
+  timeoutThreshold: schema.number(),
+  port: schema.number(),
+});
+
+export type ProxyConfigType = TypeOf<typeof createProxySchema>;
+
+export class ProxyConfig {
```

Comment:
ConfigClass is not required anymore https://github.com/elastic/kibana/blob/master/src/core/MIGRATION.md#configure-plugin

Reply:
Oh this greatly simplifies this!

Reply:
However! I'm not sure how to use this since the config is being set in the constructor as a member property of the proxy  class, but i have no interface to provide to `Observable` :

`  private readonly config$: Observable<ProxyConfig>;`

or the signature for `setConfig` which is `setConfig(config: ProxyConfig)`?

Reply:
`private readonly config$: Observable<ProxyConfigType>;` should do the trick. I should add this to the docs. 👍

Reply:
https://github.com/elastic/kibana/pull/38224